lib/ocamlorg_web/graphql.ml

```diff
@@ -0,0 +1,307 @@
+              get_deps info.Package.Info.conflicts)
+        ; field
+            "url"
+            ~doc:"The url of the package"
+            ~args:Arg.[]
+            ~typ:url
+            ~resolve:(fun _ p ->
+              let info = Package.info p in
+              info.Package.Info.url)
+        ]))
+
+let packages_result =
+  Graphql_lwt.Schema.(
+    obj "allPackages" ~fields:(fun _ ->
+        [ field
+            "totalPackages"
+            ~doc:"total number of packages"
+            ~args:Arg.[]
+            ~typ:(non_null int)
+            ~resolve:(fun _ p -> p.total_packages)
+        ; field
+            "packages"
+            ~doc:"list of all packages"
+            ~args:Arg.[]
+            ~typ:(non_null (list (non_null package)))
+            ~resolve:(fun _ p -> p.packages)
+        ]))
+
+let schema : Dream.request Graphql_lwt.Schema.schema =
+  Graphql_lwt.Schema.(
+    schema
+      [ field
+          "allPackages"
+          ~typ:(non_null packages_result)
+          ~doc:
+            "Filter packages by name passing search query. Packages can also \
+             be paginated by setting the offset and limit as desired"
+          ~args:
+            Arg.
+              [ arg
+                  ~doc:"Filter packages by passing a search query"
+                  "filter"
+                  ~typ:string
+              ; arg'
+                  ~doc:
+                    "Specifies at what index packages can start, set to 0 by \
+                     default which means start from the first package"
+                  "offset"
+                  ~typ:int
+                  ~default:0
+              ; arg
+                  ~doc:
+                    "Specifies the limit which means the number of packages \
+                     you want to return if you do not want all packages \
+                     returned at once. By default, all the packages are \
+                     returned"
+                  "limit"
+                  ~typ:int
+              ]
+          ~resolve:(fun _ () filter offset limit ->
+            let packages = Package.all_packages_latest () in
+            let totalPackages = List.length packages in
+            match limit with
+            | None ->
+              let limit = totalPackages in
+              let packages_result =
+                get_packages_result totalPackages offset limit filter packages
+              in
+              packages_result
+            | Some limit ->
+              let limit = limit in
+              let packages_result =
+                get_packages_result totalPackages offset limit filter packages
+              in
+              packages_result)
+      ; field
+          "package"
+          ~typ:package
+          ~doc:
+            "Returns details of a specified package. It returns the latest \
+             version if no version is specifed or returns a particular version \
+             of the package if a specified"
```

Comment:
Is it true what you say about the version? I don't find where in the code you allow to specify a version and where you handle it.

Reply:
> Is it true what you say about the version? I don't find where in the code you allow to specify a version and where you handle it.

Yea, it was there before but you guys mentioned to leave out the versions for now. So I forgot to update this line

Reply:
I think that to specify a version you only need a check if two versions are equal and not a version inequality check, so it should be straight-forward and uncontroversial to add it. So if you've already written it, I'd say it's ok to add it (@tmattio and @patricoferris , do you agree?). But I wouldn't invest time into it now.

Reply:
> Thanks for the review @pitag-ha, all sounds very reasonable to me. Once these changes land, I think we're ready to merge. Do ping me @dinakajoy when they're in :))

Hi, @patricoferris the changes has been effected.

Reply:
In the end, do you prefer to correct the `doc`? Or to add the code to handle the version?

Reply:
@dinakajoy, have you seen this? :)

Reply:
Sorry, I forgot to implement it but it will be part of my next push